### ISSUE TITLE
Incorrect literate highlighting for mathematics delimiters

### ISSUE THREAD

**Issue by simon-nicholls** [JustusAdam/language-haskell#82]
`$` is matched as a delimiter despite being positioned in an inline code span, or a following code block.

So for:

```
A Haskell type definition for $\mathbb{N}$ and test of `$`

~~~haskell
data Nat = Zero | Succ Nat
test = (++"...gnom!") <$> ["pizza", "ljubljanska"]
```

incorrect highlighting will be displayed for the following section:

```
$`

~~~haskell
data Nat = Zero | Succ Nat
test = (++"...gnom!") <$
```

When switching from Literate Haskell mode to others, for testing alternate syntax support:

Markdown All in One looks right on casual inspection, and appears only to recognise mathematics delimiters that are outside of such code blocks, plus exist in pairs on the same line for `$`, and newlines for `$$`.

For a LaTeX equivalent test, both Literate Haskell mode and LaTeX workshop exhibit the highlighting issue; `$`s are not paired solely within the current section of text / environment.

**Comment by simon-nicholls** [comment]
Just learned about the "Inspect TM Scopes" feature, and some more about the general way these extensions work.

Within the incorrectly highlighted section, the scopes are:

```
support.class.math.block.tex
text.tex.latex.haskell
```

I can see that in LaTeX Workshop, both `support.class.math.block.tex` and `text.tex.latex` are defined as expected, and disabling LaTeX Workshop shows that's where the bleedover into the Haskell subscope is happening.

So I'll raise an issue over there for this one.

**Comment by JustusAdam** [comment]
Thanks anyways for both the filing and the inspection work 👍

**Comment by simon-nicholls** [comment]
No worries.

LaTeX Workshop have already declared `code` as a generic verbatim block in master, fixing `$` use within standard literate Haskell code blocks, and I've submitted a pull request for `hscode` and `hsinline` support.

Those would be good for typesetting, as they play well with some existing naming conventions (e.g. minted & listings packages), and I think it would be great to recognise them in language-haskell (crucially `hscode`, so the Haskell language extension can hook into it). I can hack on that as things progress.

`spec` I use a lot, but I've not addressed yet. The naming I found as a loose naming convention for [hiding Haskell from the compiler](https://wiki.haskell.org/Literate_programming#Hiding_code_from_Haskell), but if current tooling is merely ignoring non-`code` blocks, then we've a chance to support something `hs` related instead, that syntax highlights, doesn't compile, and other communities would be happy to see as Haskell related.
